include/bm/bm_sim/field_lists.h

```diff
@@ -46,23 +47,42 @@ class FieldList {
     bool operator==(const field_t& other) const {
       return header == other.header && offset == other.offset;
     }
-
     bool operator!=(const field_t& other) const {
       return !(*this == other);
     }
   };
 
+  struct constant_t {
+    int value;
+    size_t nbits;
+
+    bool operator==(const constant_t& other) const {
+      return value == other.value && nbits == other.nbits;
+    }
+    bool operator!=(const constant_t& other) const {
+      return !(*this == other);
+    }
+  };
+
  public:
-  using iterator = std::vector<field_t>::iterator;
-  using const_iterator = std::vector<field_t>::const_iterator;
-  using reference = std::vector<field_t>::reference;
-  using const_reference = std::vector<field_t>::const_reference;
+  using field_list_member_t = boost::variant<field_t, constant_t>;
```

Comment:
AFAIK, boost visitors don't use virtual polymorphism. I am just reproducing this here. You could use virtual polymorphism if you wanted here and accept a `FieldListVisitor`, but you would need to make both operators in `FieldListVistor` virtual. You would also incur the cost of a virtual function call for each member of the variant vector (which is why boost visitors don't use it probably).